pkg/testmachinery/controller/reconciler/testrun_control.go

```diff
@@ -160,6 +160,16 @@ func (r *TestmachineryReconciler) createWorkflow(ctx context.Context, rCtx *reco
 	rCtx.tr.Status.Workflow = rCtx.wf.Name
 	rCtx.tr.Status.Phase = tmv1beta1.RunPhaseRunning
 
+	// update status first because otherwise it would be lost during the patch call
+	rCtx.tr.Status.ObservedGeneration = rCtx.tr.Generation
+	if err := r.Status().Update(ctx, rCtx.tr); err != nil {
+		log.Error(err, "unable to update testrun status")
+		return reconcile.Result{}, err
+	}
+
+	beforePatch := rCtx.tr.DeepCopy()
+	patch := client.MergeFrom(beforePatch)
```

Comment:
beforePatch is not required and can be dropped.
```suggestion
	patch := client.MergeFrom(rCtx.tr.DeepCopy()
)
```